sycl/include/sycl/ext/oneapi/experimental/invoke_simd.hpp

```diff
@@ -220,14 +229,23 @@ static constexpr int get_sg_size() {
 // with captures. Note __regcall - this is needed for efficient argument
 // forwarding.
 template <int N, class Callable, class... T>
-__regcall detail::SimdRetType<N, Callable, T...>
-simd_call_helper(const void *obj_ptr,
-                 typename detail::spmd2simd<T, N>::type... simd_args) {
+SYCL_EXTERNAL __regcall detail::SimdRetType<N, Callable, T...>
+simd_obj_call_helper(const void *obj_ptr,
+                     typename detail::spmd2simd<T, N>::type... simd_args) {
   auto f =
       *reinterpret_cast<const std::remove_reference_t<Callable> *>(obj_ptr);
```

Comment:
Why is this reinterpret_cast needed? (I realize it's not changed in this PR but I would like to understand)

Reply:
I think because `const void*` can't be used as a function pointer. Another question is why `obj_ptr` is `const void*` and not `const Callable*`. I tried changing (along with `__builtin_invoke_simd` declaration, got lots of errors. Will need to figure out once functors/lambdas are enabled.